store/tikv/region_cache.go

```diff
@@ -29,17 +29,25 @@ type RegionCache struct {
 	pdClient pd.Client
 	mu       sync.RWMutex
 	// TODO: store in array and use binary search
```

Comment:
@morefreeze It's possible. When inserting a new Region, we need to remove all old regions that are overlapped with the new one. I have a POC on my machine and will fix it in following PR.